### ISSUE TITLE
Check the config file before to read it

### ISSUE THREAD

**Issue by sdouche** [vaab/gitchangelog#54]
Hello,
It would be cool to check:
- if the config file exists before to read it
- if the section exists

```
$ gitchangelog
Traceback (most recent call last):
  File "/home/sdouche/.pyenv/versions/default/bin/gitchangelog", line 11, in <module>
    sys.exit(main())
  File "/home/sdouche/.pyenv/versions/3.5.1/envs/default/lib/python3.5/site-packages/gitchangelog.py", line 1030, in main
    ignore_regexps=config['ignore_regexps'],
KeyError: 'ignore_regexps'
```

**Comment by vaab** [comment]
Hi, ;-) ... 

Hmm, config file existence is checked... but bugs might exist, may I ask you to provide the scenario and the output that makes you think that when the config file doesn't exist, it won't be detected as such ? I'll be glad to tackle that.

Same question for the check of existences of sections : the assumption made by the code is that the released files are coherent and that a default value is provided by the ``gitchangelog.rc.reference``. Let me rephrase this: the code does not need to check for existence of a section as we provide a default value thanks to the ``gitchangelog.rc.reference`` that should match the one used by the code.

If something was not clear and lead you to act genuinely in a way that lead you to break your gitchangelog configuration, there might be information lacking from the documentation part. What would come to mind is that you changed your ``gitchangelog.rc.reference`` (which you should not change), and possibly commented the ``ignore_regexps``. I invite you instead to revert this file back, and simply override the value in your own ``.gitchangelog.rc``.

Otherwise, it is possible that I have not clearly understood your report...

Thanks for the feedback and your interest for this project. I'll be glad to share with you a new beer as we did last time in Paris, a lot have changed since this last time !

**Comment by vaab** [comment]
I'm closing this due to incomplete information. Feel free to answer my questions so I can identify a bug, misbehavior, or amelioration that would help you in this case.

**Comment by sdouche** [comment]
Hi,
Sorry for the lag. How to reproduce this:

```
# Creating a clean environment
$ virtualenv test
Using base prefix '/home/sdouche/.pyenv/versions/3.5.1'
New python executable in /home/sdouche/test/bin/python3.5
Also creating executable in /home/sdouche/test/bin/python
Installing setuptools, pip, wheel...done.

# Install gitchangelog
$ .test//bin/pip install gitchangelog
Collecting gitchangelog
Installing collected packages: gitchangelog
Successfully installed gitchangelog-2.3.0

# Try it on a random repo
$ cd ~/src/coreos/rkt/
$ ~/test/bin/gitchangelog 
Traceback (most recent call last):
  File "/home/sdouche/test/bin/gitchangelog", line 11, in <module>
    sys.exit(main())
  File "/home/sdouche/test/lib/python3.5/site-packages/gitchangelog.py", line 1027, in main
    ignore_regexps=config['ignore_regexps'],
KeyError: 'ignore_regexps'
```

**Comment by vaab** [comment]
I want to make ``gitchangelog`` warn about these missing keys and in the same time not require the existence of ``gitchangelog.rc.reference`` (which was at the core of your issue, with the added fact that the error messages are not clear). This seems indeed the saner option and provide much more than proper error messages. The final aim is to have a ``gitchangelog.py`` that has no dependency, and is a one-file project that could be simply copied anywhere if needed.
